src/SDK/Common/Configuration/EnvironmentResolver.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\SDK\Common\Configuration;
+
+/**
+ * @internal
+ */
+class EnvironmentResolver extends Resolver
+{
+    public function hasVariable(string $variableName): bool
+    {
+        return getenv($variableName) !== false || isset($_ENV[$variableName]);
+    }
+
+    /**
+     * @psalm-suppress InvalidReturnStatement
+     * @psalm-suppress InvalidReturnType
+     */
+    public function retrieveValue(string $variableName, ?string $default = ''): ?string
+    {
+        $value = getenv($variableName);
+        if ($value === false) {
+            $value = $_ENV[$variableName] ?? $default;
```

Comment:
`$value` retreived from `$_ENV` can be any type. I think we should allow resolvers to return `mixed`.

Reply:
That's true, but there is a bunch of logic in `Accessor` to parse and validate string values, and I think it makes sense to leave it there.

Reply:
I think we should skip parsing if `$value` is not a string and just validate/coerce the type.
Limiting the type to `string` will become an actual issue when configuration files with support for more complex types are added to the spec.

Reply:
resolvers can return `mixed` now. Rather than skip parsing, I updated the parsers to allow their own type (eg bool|string)...